storage-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -16,27 +16,65 @@
 
 package io.delta.storage;
 
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import io.delta.storage.HadoopFileSystemLogStore;
+import org.apache.commons.io.IOUtils;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
+import org.apache.commons.collections.iterators.FilterIterator;
 
 import java.io.IOException;
 import java.util.Iterator;
+import java.util.regex.Pattern;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+* Used so that we can use an external store child implementation
+* to provide the mutual exclusion that the cloud store,
+* e.g. s3, is missing.
+*/
 public abstract class BaseExternalLogStore extends HadoopFileSystemLogStore {
+    private static final Logger LOG = LoggerFactory.getLogger(BaseExternalLogStore.class);
 
     ////////////////////////
     // Public API Methods //
     ////////////////////////
 
     public BaseExternalLogStore(Configuration hadoopConf) {
         super(hadoopConf);
+        deltaFilePattern = Pattern.compile("\\d+\\.json");
+
     }
 
     @Override
     public Iterator<FileStatus> listFrom(Path path, Configuration hadoopConf) throws IOException {
-        // TODO
-        return null;
+        FileSystem fs = path.getFileSystem(hadoopConf);
+        Path resolvedPath;
+        try {
+            resolvedPath = stripUserInfo(fs.makeQualified(path));
+        } catch(java.net.URISyntaxException e) {
+            throw new IOException(e);
+        };
+        Path tablePath = getTablePath(resolvedPath);
+        ExternalCommitEntry entry = getLatestExternalEntry(tablePath);
+        if(entry != null) {
+            fixDeltaLog(fs, entry);
+        }
+
+        return new FilterIterator(
+            super.listFrom(path, hadoopConf),
+            item -> {
+                String name = ((FileStatus)item).getPath().getName();
+                return !name.startsWith("_") && !name.startsWith(".");
```

Comment:
inline docs explaining this `_` and `.`

Reply:
I just removed this extra filtering and everything seems to work (both unit tests and integration test).

The idea was to exclude this temporary directory we create in _delta_log, but it is probably not necessary, because:
- this method lists only files, not directories (in old scala code we were creating these temporary files directly in _delta_log, that's why we had to filter them out)
- this method returns filenames >= path and '.' is lexicographically smaller than '0', so file names starting with '.' should be filtered out this way.

It seems filtering out files starting with '_' is also not necessary (for example one of the files created by delta itself is _last_checkpoint)